apps/web/pages/api/users/username/[username].ts

```diff
@@ -0,0 +1,51 @@
+import type { NextApiRequest, NextApiResponse } from "next";
```

Comment:
I recommend converting this REST endpoint to a tRPC endpoint instead. This would allow us to leverage our existing authentication, rate limiting, and other middleware that's already set up in the application. It would also provide better type safety and consistency with the rest of our codebase.